src/commands/kubeconfig.ts

```diff
@@ -0,0 +1,248 @@
+**/
+import { retryWithSleep } from "../common/retry";
+import { authenticate } from "../drivers/auth";
+import { guard } from "../drivers/firestore";
+import { print2 } from "../drivers/stdio";
+import { getAwsConfig } from "../plugins/aws/config";
+import {
+  awsCloudAuth,
+  profileName,
+  requestAccessToCluster,
+  getAndValidateK8sIntegration,
+} from "../plugins/kubeconfig";
+import { ensureEksInstall } from "../plugins/kubeconfig/install";
+import { exec, stricmp } from "../util";
+import { writeAwsConfigProfile, writeAwsTempCredentials } from "./aws/files";
+import yargs from "yargs";
+
+export type KubeconfigCommandArgs = {
+  cluster: string;
+  role: string;
+  resource?: string;
+  reason?: string;
+  requestedDuration?: string;
+};
+
+export const kubeconfigCommand = (yargs: yargs.Argv) =>
+  yargs.command<KubeconfigCommandArgs>(
+    "kubeconfig",
+    "Request access to and automatically configure kubectl for a k8s cluster hosted by a cloud provider. Currently supports AWS EKS only.",
+    (yargs) =>
+      yargs
+        .option("cluster", {
+          type: "string",
+          demandOption: true,
+          describe: "The ID of the k8s cluster as configured P0 Security",
+        })
+        .option("resource", {
+          type: "string",
+          describe:
+            'The resource or resource type (e.g., "Pod / *"), or omit for all',
+        })
+        .option("role", {
+          type: "string",
+          demandOption: true,
+          describe:
+            'The k8s role to request, e.g., "ClusterRole / cluster-admin"',
+        })
+        .option("reason", {
+          type: "string",
+          describe: "Reason access is needed",
+        })
+        .option("requested-duration", {
+          type: "string",
+          // Copied from the P0 backend
+          describe:
+            "Requested duration for access (format like '10 minutes', '2 hours', '5 days', or '1 week')",
+        }),
+    guard(kubeconfigAction)
+  );
+
+const kubeconfigAction = async (
+  args: yargs.ArgumentsCamelCase<KubeconfigCommandArgs>
+) => {
+  if (!(await ensureEksInstall())) {
+    throw "Required dependencies are missing; please try again after installing them, or check that they are available on the PATH.";
+  }
+
+  const role = normalizeRoleArg(args.role);
+
+  if (args.resource) {
+    validateResourceArg(args.resource);
+  }
+
+  const authn = await authenticate();
+
+  const clusterConfig = await getAndValidateK8sIntegration(authn, args.cluster);
+
+  const { clusterId, awsAccountId, awsClusterArn } = clusterConfig;
+  const { config } = await getAwsConfig(authn, awsAccountId);
+  const { login } = config;
+
+  // Verify that the AWS auth type is supported before issuing the requests
+  if (!login?.type || login?.type === "iam") {
+    throw "This AWS account is not configured for kubectl access via the P0 CLI.\nYou can request access to the cluster using the `p0 request k8s` command.";
```

Comment:
Shouldn't this error tell the user to login using an accepted `p0 login` method?

Reply:
This is checking for the configured login type for the AWS account the EKS cluster is configured under in P0, so I don't think `p0 login` is relevant. Compare to the [equivalent behavior in AWS SSH](https://github.com/p0-security/p0cli/blob/2ae3841b9ae920cbee4f0dd895964270af85ab31/src/plugins/aws/ssh.ts#L64).